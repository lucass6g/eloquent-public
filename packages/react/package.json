{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
