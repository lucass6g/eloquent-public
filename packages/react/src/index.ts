export {Switch, Case, Default} from './components/conditional-switch'