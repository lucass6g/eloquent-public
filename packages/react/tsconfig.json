{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
