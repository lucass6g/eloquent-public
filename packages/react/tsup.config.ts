import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})