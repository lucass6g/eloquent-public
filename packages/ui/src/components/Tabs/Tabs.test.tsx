import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Tabs } from "..";
import { vi } from "vitest";

describe("Tabs", () => {
  it("should match the snapshot", () => {
    const component = render(
      <Tabs.Root defaultValue="account">
        <Tabs.List>
          <Tabs.Trigger value="account">tab trigger 1</Tabs.Trigger>
          <Tabs.Trigger value="password">tab trigger 2</Tabs.Trigger>
        </Tabs.List>
        <Tabs.Content value="account">tab content 1</Tabs.Content>
        <Tabs.Content value="password">tab content 2</Tabs.Content>
      </Tabs.Root>
    );

    expect(component.baseElement).toMatchSnapshot();
  });

  it("should change tab correctly ", async () => {
    const tabsFn = vi.fn();

    render(
      <Tabs.Root defaultValue="account" onValueChange={tabsFn}>
        <Tabs.List>
          <Tabs.Trigger value="account">tab trigger 1</Tabs.Trigger>
          <Tabs.Trigger value="password">tab trigger 2</Tabs.Trigger>
        </Tabs.List>
        <Tabs.Content value="account">tab content 1</Tabs.Content>
        <Tabs.Content value="password">tab content 2</Tabs.Content>
      </Tabs.Root>
    );

    const content1 = screen.getByText("tab content 1");
    expect(content1).toBeInTheDocument();

    const trigger = screen.getByRole("tab", { name: /tab trigger 2/i });
    await waitFor(() => userEvent.click(trigger));

    const content2 = screen.getByText("tab content 2");
    expect(content2).toBeInTheDocument();

    expect(tabsFn).toHaveBeenCalled();
  });
});
