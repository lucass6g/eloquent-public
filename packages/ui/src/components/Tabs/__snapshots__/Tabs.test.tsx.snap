// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tabs > should match the snapshot 1`] = `
<body>
  <div>
    <div
      data-orientation="horizontal"
      dir="ltr"
    >
      <div
        aria-orientation="horizontal"
        class="bg-neutral-200 rounded-[6px] h-9 p-1 text-muted-foreground"
        data-orientation="horizontal"
        role="tablist"
        style="outline: none;"
        tabindex="0"
      >
        <button
          aria-controls="radix-:r0:-content-account"
          aria-selected="true"
          class="inline-flex rounded-[2px] items-center justify-center text-neutral-500 whitespace-nowrap px-3 py-1 text-sm ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-background data-[state=active]:text-foreground data-[state=active]:font-medium"
          data-orientation="horizontal"
          data-radix-collection-item=""
          data-state="active"
          id="radix-:r0:-trigger-account"
          role="tab"
          tabindex="-1"
          type="button"
        >
          tab trigger 1
        </button>
        <button
          aria-controls="radix-:r0:-content-password"
          aria-selected="false"
          class="inline-flex rounded-[2px] items-center justify-center text-neutral-500 whitespace-nowrap px-3 py-1 text-sm ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-background data-[state=active]:text-foreground data-[state=active]:font-medium"
          data-orientation="horizontal"
          data-radix-collection-item=""
          data-state="inactive"
          id="radix-:r0:-trigger-password"
          role="tab"
          tabindex="-1"
          type="button"
        >
          tab trigger 2
        </button>
      </div>
      <div
        aria-labelledby="radix-:r0:-trigger-account"
        class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
        data-orientation="horizontal"
        data-state="active"
        id="radix-:r0:-content-account"
        role="tabpanel"
        style="animation-duration: 0s;"
        tabindex="0"
      >
        tab content 1
      </div>
      <div
        aria-labelledby="radix-:r0:-trigger-password"
        class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
        data-orientation="horizontal"
        data-state="inactive"
        hidden=""
        id="radix-:r0:-content-password"
        role="tabpanel"
        tabindex="0"
      />
    </div>
  </div>
</body>
`;
